### PR TITLE
cheesydrive

### PR DESCRIPTION
**Pull request by BBScholar**
Added cheesy drive from 254's codebase. needs to be tested before merging

### CODE REVIEW

File: src/main/kotlin/frc/team5499/frc2018Kotlin/controllers/TeleopController.kt [hunk 1]
```diff
@@ -4,21 +4,52 @@ import edu.wpi.first.wpilibj.XboxController
...
+        SPACE,
+        TANK
+    }
+
+    var driveConfig = Constants.Input.DRIVE_CONFIG
+
     val driver = XboxController(Constants.Input.DRIVER_PORT)
     val codriver = XboxController(Constants.Input.CODRIVER_PORT)
 
+    val driverHelper = DriveHelper()
```
[the remark this answers is not in the document]
Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/controllers/TeleopController.kt]: Fixed

File: src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Drivetrain.kt [hunk 2]
```diff
@@ -367,6 +368,11 @@ object Drivetrain : Subsystem() {
         mRightMaster.set(ControlMode.PercentOutput, right)
     }
 
+    fun setPercent(signal: DriveSignal) {
```
[the remark this answers is not in the document]
Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Drivetrain.kt]: I think its nice to have

**Comment by connorworley** [comment]
Please add some unit tests.

File: src/main/kotlin/frc/team5499/frc2018Kotlin/utils/Utils.kt [hunk 3]
```diff
@@ -7,6 +7,14 @@ object Utils {
     const val INCHES_PER_SECOND_PER_ENCODER_TICKS_PER_100MS = 10.0
     const val DEGREES_PER_ROTATION = 360.0
 
+    fun limit(value: Double, limit: Double): Double {
```
[the remark this answers is not in the document]
Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/utils/Utils.kt]: This isn't an issue

File: src/main/kotlin/frc/team5499/frc2018Kotlin/utils/DriveHelper.kt [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+    fun tankDrive(left: Double, right: Double, isSlow: Boolean): DriveSignal {
+        val newLeft = handleDeadband(left, kThrottleDeadband)
+        val newRight = handleDeadband(right, kThrottleDeadband)
+        val mult = if (isSlow) kSlowMult else 1.0
+        return DriveSignal(newLeft * mult, newRight * mult)
+    }
+
+    @Suppress("LongMethod", "ComplexMethod")
+    fun cheesyDrive(throttle: Double, wheel: Double, isQuickTurn: Boolean, isHighGear: Boolean): DriveSignal {
```
[the remark this answers is not in the document]
Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/utils/DriveHelper.kt]: I broke some of it up. More?

[the remark this answers is not in the document]
Comment (andycate) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/utils/DriveHelper.kt]: its gucci

File: src/main/kotlin/frc/team5499/frc2018Kotlin/utils/Utils.kt [hunk 5]
```diff
@@ -21,7 +29,7 @@ object Utils {
     }
 
     fun encoderTicksToInches(ticks: Int): Double {
-        return (Constants.WHEEL_CIR / Constants.ENCODER_TICKS_PER_ROTATION) * ticks
+        return (Constants.WHEEL_CIR / Constants.ENCODERS_TICKS_PER_ROTATION) * ticks
```
Comment (andycate) [inline src/main/kotlin/frc/team5499/frc2018Kotlin/utils/Utils.kt]: change this back to `ENCODER_TICKS_PER_ROTATION`

Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/utils/Utils.kt]: ok

**Pull request merged**